packages/gatsby-source-shopify/src/processors/product-variants.ts

```diff
@@ -0,0 +1,26 @@
+import { NodeInput } from "gatsby"
+import { pattern as idPattern } from "../node-builder"
+
+export function productVariantsProcessor(
+  objects: BulkResults,
+  builder: NodeBuilder
+): Array<Promise<NodeInput>> {
+  const objectsToBuild = objects.filter(obj => {
+    const [, remoteType] = obj.id.match(idPattern) || []
+
+    return remoteType !== `Product`
+  })
```

Comment:
Do we have to check if it's not a `Product` because we can't query for product variants directly and the new Productvariant query query's for products and product variants?

Reply:
No and yes.

We _can_ query for variants at the top level, but I found that it didn't work quite right for incremental builds. It seems that when you add a variant, querying for products that are `updated_at >= last_build_time` pulls them back, but when you query for variants there's no `created_at` filter, and `updated_at >= last_build_time` doesn't pull it back.

So because of that, I wrapped our new variants query in a products query that takes the same filters as the products query does, and now we get back all the correct records. But as a result, we get all these `Product` records in the results that we don't need because we already created nodes for them.

I suspect that merging this PR might mean that cold builds are a little bit slower. Hopefully not much. But without doing this, we won't be able to add `presentmentPrices` without removing something else, so I think we have to.

Reply:
Great, that's kind of what I figured the tradeoff would be, thanks for the explanation.